src/lib/type_check.ml

```diff
@@ -4630,6 +4630,14 @@ let check_fundef_lazy env def_annot (FD_aux (FD_function (recopt, tannot_opt, fu
     | Some vs_l -> check_tannot_opt ~def_type:"function" vs_l env vtyp_ret tannot_opt
     | None -> ()
   end;
+  (* Check $[test] functions have type unit -> unit. *)
+  (* TODO: Does the annotation go on the type declaration or the function definition? *)
```

Comment:
I think in general we put these annotations on the `val` type declarations. That's where `$[property]` and `$[counterexample]` are expected to be.

If you have a function with an inline annotation:
```
$[annot]
function f(x : int) -> unit = ...
```
it'll get expanded to
```
$[annot]
val f : int -> unit
$[annot]
function f(x) = ...
```
with the annotation on both.